app/assets/javascripts/discourse-common/addon/helpers/d-icon.js

```diff
@@ -1,7 +1,14 @@
+import { helper } from "@ember/component/helper";
 import { htmlSafe } from "@ember/template";
 import { registerUnbound } from "discourse-common/lib/helpers";
 import { renderIcon } from "discourse-common/lib/icon-library";
 
-registerUnbound("d-icon", function (id, params) {
+function icon(id, params) {
   return htmlSafe(renderIcon("string", id, params));
+}
+
+registerUnbound("d-icon", icon);
+
+export default helper(function (params, hash) {
+  return icon(...params, hash);
 });
```

Comment:
Not sure what's the history/motivation behind the `registerUnbound` stuff, this is not exactly the same thing (it's not "unbound" for one thing) but I'm not sure if that distinction really matters?

Reply:
I assume the 'unbound' part of it was originally introduced as a performance optimization. It's possible that with with modern Ember, it is no longer required.

Also worth noting that `registerUnbound` makes the helper available to our 'raw handlebars' rendering system

https://github.com/discourse/discourse/blob/ab1bb0cfa02c7f3d7db3c5805b5cc5fe1603fced/app/assets/javascripts/discourse-common/addon/lib/helpers.js#L105-L105

But I see you kept the `registerUnbound` call, so that should still work fine 👌